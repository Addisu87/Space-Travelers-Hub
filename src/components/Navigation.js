import React from 'react';
import { NavLink } from 'react-router-dom';
import planet from '../images/planet.png';
import './styles/Navigation.css';

const Navigation = () => (
  <>
    <nav className="navbar">
      <div className="navTitle">
        <img id="logo" src={planet} alt="Logo" />
        <h1>Space Travelers Hub</h1>
      </div>
      <ul className="navLinks">
        <li className="navLink">
          <NavLink to="/">Rockets</NavLink>
        </li>
        <li className="navLink">
          <NavLink to="/missions">Missions</NavLink>
        </li>
        <div id="separator" />
        <li className="navLink">
          <NavLink to="/profile">My Profile</NavLink>
        </li>
      </ul>
    </nav>
    <div id="horizontal" />
  </>
);

export default Navigation;
