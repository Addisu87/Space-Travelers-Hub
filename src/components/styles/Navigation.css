.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.navTitle {
  display: flex;
  gap: 0.5rem;
  align-items: center;
}

#logo {
  width: 6%;
  height: 6%;
}

.navLinks {
  display: flex;
  gap: 0.5rem;
}

a {
  color: var(--blue);
}

.navLink:hover {
  text-decoration: underline;
}

#separator {
  border-right: 1px solid black;
}

#horizontal {
  width: 100%;
  height: 2px;
  background: #e8e8e8;
  margin: 0.5rem auto;
}
