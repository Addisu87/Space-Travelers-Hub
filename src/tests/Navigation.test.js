import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Navigation from '../components/Navigation';

describe('Jest Snapshot testing suite', () => {
  it('Matches DOM Snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
