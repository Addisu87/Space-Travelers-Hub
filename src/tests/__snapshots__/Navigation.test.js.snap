// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<nav
  class="navbar"
>
  <div
    class="navTitle"
  >
    <img
      alt="Logo"
      id="logo"
      src="planet.png"
    />
    <h1>
      Space Travelers Hub
    </h1>
  </div>
  <ul
    class="navLinks"
  >
    <li
      class="navLink"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Rockets
      </a>
    </li>
    <li
      class="navLink"
    >
      <a
        class=""
        href="/missions"
      >
        Missions
      </a>
    </li>
    <div
      id="separator"
    />
    <li
      class="navLink"
    >
      <a
        class=""
        href="/profile"
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
